firedrake/projection.py

```diff
@@ -127,8 +128,16 @@ def __init__(self, v, v_out, solver_parameters=None, constant_jacobian=True):
                              v_out.function_space())
         self.v = v
         self.v_out = v_out
-
-        if not self._same_fspace:
+        if bcs:
+            try:
+                bcs = tuple(bcs)
+            except TypeError:
+                bcs = (bcs,)
+            if any(bc.function_space() != v_out.function_space() for bc in bcs):
+                raise ValueError("bcs must be enforced on the space of the result function.")
+        self.bcs = bcs
+
+        if not self._same_fspace or self.bcs:
```

Comment:
I thought about that. But I wasn't sure if it's really the right thing to do. Wouldn't exterior BCs influence the function values between the interior and exterior nodes? I'll experiment with it and update if it turns out to be the right thing.